nservicebus/messaging/databus/index.md

```diff
@@ -19,9 +19,39 @@ Instead of serializing the payload along with the rest of the message, the `Data
 
 If the location is not available upon sending, the send operation will fail. When a message is received and the payload location is not available, the receive operation will fail as well, resulting in the standard NServiceBus retry behavior, possibly resulting in the message being moved to the error queue if the error could not be resolved.
 
-## Alternative
-
-The [Handling large stream properties via pipeline](/samples/pipeline/stream-properties/) sample demonstrates a purely stream-based approach (rather than loading the full payload into memory) implemented by leveraging the NServiceBus pipeline.
+## Transport message size limits
+
+Using the Data Bus is required when the message size can exceed the transport message size limit.
+
+Note: Not all transports have very restrictive message size limits and Azure Service Bus has increased its size limits over the years. Check the respective transport website documentation for the latest maximum limit of the message size.
+
+| Transport                  | Maximum size |
+| -------------------------- | ------------:|
+| Amazon SQS                 | 256KB        |
+| Amazon SQS + S3            | 2GB          |
+| Azure Storage Queues       | 64KB         |
+| Azure Service Bus Standard | 256KB        |
+| Azure Service Bus Premium  | 100MB        |
+| RabbitMQ                   | No limit     |
+| SQL Server                 | No limit     |
+| Learning                   | No limit     |
+| MSMQ                       | 4MB          |
+
+## Alternatives
+
+- Use a different transport or a different transport tier
+  
+- Message Compression: Use message body compression which works well on text-based payloads like XML and Json or any payload (text or binary) that contains repetitive data
+  - [Message mutator example demonstrating message body compression](/samples/messagemutators/)
+  - Community maintained [NServiceBus.Comression](https://github.com/ramonsmits/NServiceBus.Compression) with configurable compression thresholds
+- Binary Serializer: Implementing a binary serializer requires a few lines of code.
```

Comment:
couldn't you consider a binary serializer also just a form of message compression?

Reply:
> couldn't you consider a binary serializer also just a form of message compression?

Its more efficient but it's not doing any kind of compression. That would be saying Json is compression over XML.

Reply:
well technically JSON is more compact than XML 🤣